[YUNIKORN-775] Create github action to build web UI

kobe860219 opened the pull request:
### What is this PR for?
Add a github action workflow file to replace travis with github action. Unit test, build test and coverage test are included in this workflow.


### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-775

### How should this be tested?
https://app.travis-ci.com/github/apache/incubator-yunikorn-web/builds/234829228

### Screenshots (if appropriate)
<img width="1440" alt="截圖 2021-08-09 下午6 11 46" src="https://user-images.githubusercontent.com/48027290/128691994-19cb38e8-f7da-483a-bdd4-7488af141dac.png">


### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

kobe860219 commented:
@yangwwei  This workflow not run e2e test. Maybe we could skip it because it passed in the travis test and we won't use e2e for testing.

yangwwei commented on .github/workflows/main.yaml:
```diff
@@ -0,0 +1,37 @@
+        uses: actions/setup-node@v2
+        with:
+          node-version: 12.14
+      - run: npm install -g @angular/cli  > /dev/null
+      - run: npm install karma
+      - run: npm install -g protractor
+      - run: yarn install
```
I think this can be simplified to:

```
run: |
      npm install -g @angular/cli  > /dev/null
      npm install karma
      ...
```
the above syntax is supported in github action for multi-line commands.

yangwwei commented on .github/workflows/main.yaml:
```diff
@@ -0,0 +1,37 @@
+      - name: Unit Test
+        run: ng test --watch=false
+
+      - name: Build Test
+        run: ng build
```
This should be "Build", not the "Build test"

yangwwei reviewed (changes requested):
Another question is, I see in https://github.com/apache/incubator-yunikorn-web/blob/master/.travis.yml, we run 

```
make build-prod
```

and also the lint check
```
yarn lint
```

why we do not run those commands in this PR?

yangwwei commented:
Also we need to remove `.travis.yml` file in this repo

kobe860219 commented:
@yangwwei Hi, I removed .travis.yaml file and added lint test. In "Build" stage of github action workflow, it will run `ng build --prod` as run `make build:prod` in travis.

kobe860219 commented:
And I modified the "syntax" and "name of build" .

yangwwei reviewed (approved):
Looks good, thanks!
